### PR TITLE
luajit: bump new version

### PR DESCRIPTION
**Pull request by Buristan**
* test: set dependencies in BuildTestCLib macro
* Add 'cc' file type for saving bytecode.
* Fix C file generation in jit.bcsave.
* Throw any errors before stack changes in trace stitching.
* Fix recording of __concat metamethod.
* Check frame size limit before returning to a lower frame.
* build: purge sysprof.collapse module
* build: fix tool components handling
* memprof: refactor `heap_chunk` data structure
* memprof: remove unused arguments
* memprof: introduce the `--human-readable` option
* profilers: introduce event reader module
* ci: extend tarantool integration testing

Part of #9595
Part of #5994
Follows up #8700
Needed for #9217

NO_DOC=LuaJIT submodule bump
NO_TEST=LuaJIT submodule bump

### CODE REVIEW

**Comment by mkokryashkin** [comment]
I believe it's not "Part of #5994", but rather "Resolves #5994"

**Review by mkokryashkin — APPROVED** [review]

**Comment by Buristan** [comment]
@mkokryashkin, it's "Part of #5994", since #9217 is a part of it. See [the following comment](https://github.com/tarantool/tarantool/issues/9217#issuecomment-1822839641). Both will be closed together.

**Review by ylobankov — APPROVED** [review]

File: changelogs/unreleased/gh-5994-memprof-human-readable.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+## feature/tools
+
+Added the `--human-readable` option for `misc.memprof` parser to print sizes
+like 1KiB 234MiB 2GiB etc.
```
Comment (xuniq) [inline changelogs/unreleased/gh-5994-memprof-human-readable.md:4]:
```suggestion
Added the `--human-readable` option for the `misc.memprof` parser to print sizes
like 1KiB, 234MiB, 2GiB, etc.
```

File: changelogs/unreleased/gh-9595-luajit-fixes.md [hunk 2]
```diff
@@ -12,3 +12,9 @@ were fixed as part of this activity:
 * Fixed recording of `setmetatable()` with `nil` as the second argument.
 * Fixed recording of `select()` in case with negative first argument.
 * Fixed use-def analysis for child upvalues.
+* Added `cc` file type for saving bytecode.
```
Comment (xuniq) [inline changelogs/unreleased/gh-9595-luajit-fixes.md:15]:
```suggestion
* Added the `cc` file type for saving bytecode.
```

**Review by xuniq — APPROVED** [review]

File: changelogs/unreleased/gh-5994-memprof-human-readable.md (hunk 1 above)
Comment (Buristan) [reply changelogs/unreleased/gh-5994-memprof-human-readable.md]: Fixed, thanks!

File: changelogs/unreleased/gh-9595-luajit-fixes.md (hunk 2 above)
Comment (Buristan) [reply changelogs/unreleased/gh-9595-luajit-fixes.md]: Fixed, thanks!

**Pull request merged**